designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/InputUnitInterval.kt

```diff
@@ -0,0 +1,39 @@
+package org.hisp.dhis.mobile.ui.designsystem.component
+
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.text.input.ImeAction
+import androidx.compose.ui.text.input.KeyboardType
+import org.hisp.dhis.mobile.ui.designsystem.component.internal.RegExValidations
+
+@Composable
+fun InputUnitInterval(
+    title: String,
+    state: InputShellState,
+    supportingText: List<SupportingTextData>? = null,
+    legendData: LegendData? = null,
+    inputText: String? = null,
+    isRequiredField: Boolean = false,
+    onNextClicked: (() -> Unit)? = null,
+    onValueChanged: ((String?) -> Unit)? = null,
+    onFocusChanged: ((Boolean) -> Unit)? = null,
```

Comment:
Whe don't need to expose the on focusChange, onvalueChange is just fine

Reply:
@andresmr done.